controller.go

```diff
@@ -432,419 +407,261 @@ func (c *Controller) syncHandler(key string) error {
+				"firewall-name": firewallRuleName,
+				"firewall-port": strconv.Itoa(inletsPort),
+			},
+		}
 
-			if err != nil {
-				return err
-			}
+	} else if c.infraConfig.Provider == "ec2" {
 
-			if host.Status == provision.ActiveStatus {
-				if host.IP != "" {
-					err := c.updateTunnelProvisioningStatus(tunnel, provision.ActiveStatus, host.ID, host.IP)
-					if err != nil {
-						return err
-					}
+		inletsPort := inletsControlPort
 
-					err = c.updateService(tunnel, host.IP)
-					if err != nil {
-						log.Printf("Error updating service: %s, %s", tunnel.Spec.ServiceName, err.Error())
-						return fmt.Errorf("tunnel update error %s", err)
-					}
-				}
-			}
-		} else {
-			return fmt.Errorf("unsupported provider: %s", c.infraConfig.Provider)
+		if c.infraConfig.UsePro() {
+			inletsPort = inletsProControlPort
 		}
-		break
-	case provision.ActiveStatus:
-		if tunnel.Spec.ClientDeploymentRef == nil {
-			get := metav1.GetOptions{}
-			service, getServiceErr := c.kubeclientset.CoreV1().Services(tunnel.Namespace).Get(context.Background(), tunnel.Spec.ServiceName, get)
 
-			if getServiceErr != nil {
-				return getServiceErr
-			}
+		host = provision.BasicHost{
+			Name:     tunnel.Name,
+			OS:       "ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20191114",
+			Plan:     "t3.micro",
+			UserData: base64.StdEncoding.EncodeToString([]byte(userData)),
+			Additional: map[string]string{
+				"inlets-port": strconv.Itoa(inletsPort),
+			},
+		}
 
-			firstPort := int32(80)
+	} else if c.infraConfig.Provider == "civo" {
 
-			for _, port := range service.Spec.Ports {
-				if port.Name == "http" {
-					firstPort = port.Port
-					break
-				}
-			}
+		host = provision.BasicHost{
+			Name:       tunnel.Name,
+			OS:         "811a8dfb-8202-49ad-b1ef-1e6320b20497",
+			Plan:       "g2.small",
+			Region:     c.infraConfig.Region,
+			UserData:   userData,
+			Additional: map[string]string{},
+		}
 
-			ports := getPortsString(service)
+	} else if c.infraConfig.Provider == "linode" {
 
-			client := makeClient(tunnel, firstPort,
-				c.infraConfig.GetInletsClientImage(),
-				c.infraConfig.UsePro(),
-				ports,
-				c.infraConfig.ProConfig.License,
-				c.infraConfig.MaxClientMemory)
+		host = provision.BasicHost{
+			Name:       tunnel.Name,
+			OS:         "linode/ubuntu16.04lts", // https://api.linode.com/v4/images
+			Plan:       "g6-nanode-1",           // https://api.linode.com/v4/linode/types
+			Region:     c.infraConfig.Region,
+			UserData:   userData,
+			Additional: map[string]string{},
+		}
 
-			deployment, createDeployErr := c.kubeclientset.AppsV1().
-				Deployments(tunnel.Namespace).
-				Create(context.Background(), client, metav1.CreateOptions{})
+	}
+	return host
+}
 
-			if createDeployErr != nil {
-				log.Println(createDeployErr)
-			}
+func getProvisioner(c *Controller) (provision.Provisioner, error) {
+	var err error
+	var provisioner provision.Provisioner
+	if c.infraConfig.Provider == "packet" {
```

Comment:
why not a switch statement? (just a question)

Reply:
Updated to switch. Switch is actually better, I didn't change it because this `if and else` code was copied from original code so didn't really think about using switch. Thanks for pointing this out!

Reply:
I find the if statements clearer, it's my personal style, but sometimes change from if - when there's > 3-4 options.